src/main/java/hudson/plugins/git/GitStatus.java

```diff
@@ -165,25 +165,31 @@ public void writeBody(PrintWriter w) {
     public static abstract class Listener implements ExtensionPoint {
 
         /**
-         * Called when there is a change notification on a specific repository url.
-         *
-         * @param uri      the repository uri.
-         * @param branches the (optional) branch information.
-         * @return any response contributors for the response to the push request.
-         * @since 1.4.1
-         * @deprecated implement #onNotifyCommit(org.eclipse.jgit.transport.URIish, String, String...)
+         * @deprecated implement {@link #onNotifyCommit(org.eclipse.jgit.transport.URIish, String, List, String...)}
          */
         public List<ResponseContributor> onNotifyCommit(URIish uri, String[] branches) {
-            return onNotifyCommit(uri, null, branches);
+            throw new AbstractMethodError();
```

Comment:
Here we are assuming that

* the only legitimate callers are `GitStatus` itself
* original implementations must have implemented this method (perhaps originally it was `abstract`)